### PR TITLE
Clave integration with custom msg builder

### PR DESCRIPTION
**Pull request by tramuntanal**
This PR adds integration with Cl@ve authentication system.
Because of the specificities of Cl@ve it uses a custom request message builder to forge requests to the Cl@ve system.

### CODE REVIEW

File: config/secrets.yml [hunk 1]
```diff
@@ -31,6 +31,15 @@ default: &default
       enabled: false
       client_id: <%= ENV["OMNIAUTH_GOOGLE_CLIENT_ID"] %>
       client_secret: <%= ENV["OMNIAUTH_GOOGLE_CLIENT_SECRET"] %>
+    clave:
+      enabled: true
+      # assertion_consumer_service_url: <%= ENV["CLAVE_ASSERTION_CONSUMER_SERVICE_URL"] %>
```
Comment (laurajaime) [inline config/secrets.yml]: is this key necessary?

File: config/initializers/omniauth.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+               conf[:attribute_statements] = {
+                 name: ["http://eidas.europa.eu/attributes/naturalperson/CurrentGivenName", "name"],
+                 email: %w(email mail),
+                 first_name: ["http://eidas.europa.eu/attributes/naturalperson/CurrentGivenName", "first_name", "firstname", "firstName"],
+                 first_surname: ["http://es.minhafp.clave/FirstSurname", "first_name", "firstname", "firstName"],
+                 last_name: ["http://eidas.europa.eu/attributes/naturalperson/CurrentFamilyName", "last_name", "lastname", "lastName"]
+
+               }
+             },
+      scope: :autenticacio_usuari
```
Comment (laurajaime) [inline config/initializers/omniauth.rb:73]: Is this scope name correct? 🤔

**Review by laurajaime — COMMENTED** [review]
Great job!

Only two details.

File: config/initializers/omniauth.rb (hunk 2 above)
Comment (tramuntanal) [reply config/initializers/omniauth.rb:73]: yes

**Review by laurajaime — APPROVED** [review]
LGTM!

**Pull request merged**